cmd/stack/proxy.go

```diff
@@ -0,0 +1,319 @@
+	store := fctl.GetOrganizationStore(cmd)
+	c.profile = store.Config.GetProfile(fctl.GetCurrentProfileName(cmd, store.Config))
+
+	organizationID, err := fctl.ResolveOrganizationID(cmd, store.Config, store.Client())
+	if err != nil {
+		return nil, err
+	}
+
+	stack, err := fctl.ResolveStack(cmd, store.Config, organizationID)
+	if err != nil {
+		return nil, err
+	}
+
+	c.store.OrganizationID = organizationID
+	c.store.StackID = stack.Id
+	c.store.StackUrl = c.profile.ServicesBaseUrl(stack).String()
+
+	port := fctl.GetInt(cmd, proxyPortFlag)
+	c.store.ProxyUrl = fmt.Sprintf("http://localhost:%d", port)
+
+	stackBaseURL, err := url.Parse(c.store.StackUrl)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing stack URL: %v", err)
+	}
+
+	fmt.Fprintf(cmd.OutOrStdout(), "Target Stack URL: %s\n", stackBaseURL.String())
+
+	proxy := httputil.NewSingleHostReverseProxy(stackBaseURL)
+
+	originalDirector := proxy.Director
+	proxy.Director = func(req *http.Request) {
+		originalDirector(req)
+		req.Host = stackBaseURL.Host
+
+		targetURL := &url.URL{
+			Scheme:   stackBaseURL.Scheme,
+			Host:     stackBaseURL.Host,
+			Path:     req.URL.Path,
+			RawQuery: req.URL.RawQuery,
+		}
+
+		sourceURL := &url.URL{
+			Path:     req.URL.Path,
+			RawQuery: req.URL.RawQuery,
+		}
+
+		fmt.Fprintf(cmd.OutOrStdout(), "[%s] Proxying %s %s to %s\n",
+			time.Now().Format(time.RFC3339),
+			req.Method,
+			sourceURL.String(),
+			targetURL.String())
+	}
+
+	proxy.ErrorHandler = func(w http.ResponseWriter, r *http.Request, err error) {
+		targetURL := &url.URL{
+			Scheme:   stackBaseURL.Scheme,
+			Host:     stackBaseURL.Host,
+			Path:     r.URL.Path,
+			RawQuery: r.URL.RawQuery,
+		}
+
+		fmt.Fprintf(cmd.ErrOrStderr(), "[%s] ERROR proxying request to %s: %v\n",
+			time.Now().Format(time.RFC3339),
+			targetURL.String(),
+			err)
+
+		w.Header().Set("Content-Type", "text/plain")
+		w.WriteHeader(http.StatusBadGateway)
+		errorMsg := fmt.Sprintf("Proxy Error: %v\nTarget URL: %s",
+			err,
+			targetURL.String())
+		io.WriteString(w, errorMsg)
+	}
+
+	transport := &tokenTransport{
+		wrapped:  http.DefaultTransport,
+		profile:  c.profile,
+		stack:    stack,
+		cmd:      cmd,
+		tokenMux: &sync.RWMutex{},
+	}
+
+	proxy.Transport = transport
+
+	server := &http.Server{
+		Addr: fmt.Sprintf(":%d", port),
+		Handler: http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			proxy.ServeHTTP(w, r)
+		}),
+		ReadHeaderTimeout: 10 * time.Second,
+	}
+
+	stop := make(chan os.Signal, 1)
+	signal.Notify(stop, os.Interrupt, syscall.SIGTERM)
+
+	go func() {
+		fmt.Fprintf(cmd.OutOrStdout(), "Starting proxy server at %s -> %s\n", c.store.ProxyUrl, c.store.StackUrl)
+		fmt.Fprintf(cmd.OutOrStdout(), "Press Ctrl+C to stop the server\n")
+
+		if err := server.ListenAndServe(); err != nil && err != http.ErrServerClosed {
+			fmt.Fprintf(cmd.ErrOrStderr(), "Server error: %v\n", err)
+			os.Exit(1)
```

Comment:
It's the role of the cobra cli to exit, you should not exit like that in a goroutine.
You can use an error channel to forward the error and handle it in the main routine.